### PR TITLE
Add fixes to BSplineCurve2D.line_intersections, and some other fixes

### PR DESCRIPTION
**Pull request by WirajanDASILVA**
…ixes

* **Please check if the PR fulfills these requirements**
- [X] The commit messages follows our guidelines (explicit, in english)
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [X] Bug fix:
- [ ] New features:
- [ ] Documentation
- [ ] Optimization

* **Other information**:

### CODE REVIEW

File: volmdlr/edges.py [hunk 1]
```diff
@@ -1379,29 +1393,37 @@ def rotation_inplace(self, center: volmdlr.Point2D, angle: float):
...
+                        n for n in npy.linspace(initial_abscissa, initial_abscissa + linesegment.length(),
+                                                number_points))
                 else:
-                    list_abcissas = [initial_abscissa * n for n in
-                                     npy.linspace(0.9, 1, 100)]
+                    resolution = 1e-6
+                    nb_pts = max(2, int(0.0001 / resolution))
+                    list_abcissas = list(new_abscissa for new_abscissa in npy.linspace(abscissa - 0.0001 / 2,
+                                                                                       abscissa + 0.0001 / 2, nb_pts))
                 distance = npy.inf
```
Comment (GabrielJMS) [inline volmdlr/edges.py:1416]: maybe it's a good idea to put 0.0001 as a variable, so we can easily adjust if we need to debug someday.

File: volmdlr/wires.py [hunk 2]
```diff
@@ -4238,13 +4201,12 @@ def frame_mapping(self, frame: volmdlr.Frame2D, side: str):
         if side == 'old':
             return Ellipse2D(self.major_axis, self.minor_axis, frame.old_coordinates(self.center),
```
Comment (GabrielJMS) [inline volmdlr/wires.py:4202]:
```suggestion
            return Ellipse2D(self.major_axis, self.minor_axis, frame.local_to_global_coordinates(self.center),
```

File: volmdlr/wires.py [hunk 3]
```diff
@@ -4238,13 +4201,12 @@ def frame_mapping(self, frame: volmdlr.Frame2D, side: str):
...
                              self.major_dir)
-        elif side == 'new':
+        if side == 'new':
             point_major_dir = self.center + self.major_dir * self.major_axis
             major_dir = frame.new_coordinates(point_major_dir) - self.center
             return Ellipse2D(self.major_axis, self.minor_axis, frame.new_coordinates(self.center),
```
Comment (GabrielJMS) [inline volmdlr/wires.py:4207]:
```suggestion
            return Ellipse2D(self.major_axis, self.minor_axis, frame.global_to_local_coordinates(self.center),
```

File: volmdlr/edges.py [hunk 4]
```diff
@@ -1379,29 +1393,37 @@ def rotation_inplace(self, center: volmdlr.Point2D, angle: float):
...
+                    list_abcissas = list(
+                        n for n in npy.linspace(initial_abscissa, initial_abscissa + linesegment.length(),
+                                                number_points))
                 else:
-                    list_abcissas = [initial_abscissa * n for n in
-                                     npy.linspace(0.9, 1, 100)]
+                    resolution = 1e-6
+                    nb_pts = max(2, int(0.0001 / resolution))
+                    list_abcissas = list(new_abscissa for new_abscissa in npy.linspace(abscissa - 0.0001 / 2,
+                                                                                       abscissa + 0.0001 / 2, nb_pts))
```
Comment (GabrielJMS) [inline volmdlr/edges.py:1415]: Maybe using range instead of numpy.linspace is faster?

**Review by masfaraud — DISMISSED** [review]

**Pull request closed without merging**